src/GNDStk/gnds-tree.hpp

```diff
@@ -0,0 +1,644 @@
+
+// -----------------------------------------------------------------------------
+// Tree (templated)
+// tree = Tree<>
+// -----------------------------------------------------------------------------
+
+template<
+   template<class...> class MCON, // metadata container
+   template<class...> class CCON  // children container
+>
+class Tree {
+   static std::string static_str1;
+   static std::string static_str2;
+
+public:
+
+   // ------------------------
+   // Data
+   // ------------------------
+
+   // root
+   std::shared_ptr<Node<MCON,CCON>> root;
+
+
+   // ------------------------
+   // Miscellaneous functions
+   // ------------------------
+
+   // clear
+   void clear()
+   {
+      // smart pointer, so the rest of the tree is deleted as well
+      root = nullptr;
+   }
+
+   // empty
+   bool empty() const
+   {
+      return root == nullptr;
+   }
+
+   // start
+   Node<MCON,CCON> &start(
```

Comment:
It bootstraps a brand-new GNDS hierarchy that someone wants to build. Basically, "start" a hierarchy. param1 and param2 aren't normally names that I'd used, precisely because they're non-descriptive, but there was a reason I used them here. I want to do some tweaks with the "build a GNDS tree yourself" work anyway, so it's likely that some details will change.

Reply:
I may call it init(), say, or initialize(). Or maybe start() will be just fine. I'll be sure to put in a comment. (Of course, begin() wouldn't be a good choice, as that has a different meaning throughout the whole C++ container world.)